paymentsheet/src/test/java/com/stripe/android/paymentsheet/ui/PaymentMethodFormTest.kt

```diff
@@ -0,0 +1,66 @@
+package com.stripe.android.paymentsheet.ui
+
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.test.junit4.createComposeRule
+import com.google.common.truth.Truth.assertThat
+import com.stripe.android.model.PaymentMethod
+import com.stripe.android.paymentsheet.forms.FormFieldValues
+import com.stripe.android.paymentsheet.model.PaymentSelection
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.flow.MutableStateFlow
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@OptIn(FlowPreview::class)
+@RunWith(RobolectricTestRunner::class)
+class PaymentMethodFormTest {
+
+    @get:Rule
+    val composeTestRule = createComposeRule()
+
+    @Test
+    fun changingPaymentMethodCodes_emitsOnFormFieldValuesChangedCorrectly() {
```

Comment:
Maybe we can be more specific?
```suggestion
    fun `Changing payment method emits only form values of newly selected payment method`() {
```